Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -1938,6 +1950,26 @@ public async Task UpdateImagesAsync(BaseItem item, bool forceUpdate = false)
                 }
             }
 
+            // Remove duplicate images
+            var metadataPath = item.GetInternalMetadataPath();
+            var distinctImages = imagesWithPaths
+                                    // Prefer extermal images
+                                    .OrderBy(i => !i.Path.StartsWith(metadataPath, StringComparison.Ordinal))
+                                    .DistinctBy(i => i.FileHash);
+            var imagesToRemove = imagesWithPaths.Where(i => !string.IsNullOrEmpty(i.FileHash)).Except(distinctImages).ToList();
+            if (imagesToRemove.Count > 0)
+            {
+                foreach (var image in imagesToRemove)
+                {
```

Comment:
i would put that in a try catch and aggregate all exceptions to delete as much as possible in the even of an error